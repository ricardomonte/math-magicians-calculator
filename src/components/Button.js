import PropTypes from 'prop-types';

function Button({ name, onClick }) {
  return (
    <button type="button" onClick={() => onClick(name)}>{ name }</button>
  );
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,

};

export default Button;
