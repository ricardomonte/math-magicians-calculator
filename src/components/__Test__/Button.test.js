import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Button from '../Button';

const myMockClick = jest.fn();

test('return content from Button matches snapshot', () => {
  const content = render(<Button name="8" onClick={myMockClick} />);
  expect(content.container).toMatchSnapshot();
  expect(screen.getByText('8')).toBeInTheDocument();
});
