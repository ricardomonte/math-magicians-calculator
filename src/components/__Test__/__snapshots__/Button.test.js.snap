// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return content from Button matches snapshot 1`] = `
<div>
  <button
    class="btn h-100 bg-button-normal w-25"
    type="button"
  >
    8
  </button>
</div>
`;
